WordPress/src/main/java/org/wordpress/android/ui/media/MediaAddFragment.java

```diff
@@ -102,6 +118,9 @@ public void onReceive(Context context, Intent intent) {
                     ToastUtils.showToast(context, errorMessage, ToastUtils.Duration.SHORT);
                 }
                 mCallback.onMediaAdded(mediaId);
+            } else {
+                // Coming from zero connection. Re-register upload intent.
+                onResume();
```

Comment:
Explicitly invoking lifecycle events is not recommended. Could you move whatever functionality you need into a method and use that in both places?